multicluster/controllers/multicluster/member/stale_controller.go

```diff
@@ -375,80 +366,155 @@ func (c *StaleResCleanupController) cleanupClusterInfoResourceExport(ctx context
 	return nil
 }
 
-func (c *StaleResCleanupController) cleanupMemberClusterAnnounces(ctx context.Context) error {
-	memberClusterAnnounceList := &mcsv1alpha1.MemberClusterAnnounceList{}
-	if err := c.List(ctx, memberClusterAnnounceList, &client.ListOptions{}); err != nil {
+// Run starts the MemberStaleResCleanupController and blocks until stopCh is closed.
+func (c *MemberStaleResCleanupController) Run(stopCh <-chan struct{}) {
+	klog.InfoS("Starting MemberStaleResCleanupController")
+	defer klog.InfoS("Shutting down MemberStaleResCleanupController")
+
+	ctx, _ := wait.ContextForChannel(stopCh)
```

Comment:
Do we still respect `stopCh` in the current implementation? Will all routinues/retries stop when `stopCh` is closed?

Reply:
Yeah, we need it, otherwise, the Run() will return immediately after it's finished, then there is no way to check the if there is a new event on the CommonAreaCreationCh.

Reply:
Could you answer this question:
> Will all routinues/retries stop when stopCh is closed?

Reply:
Yes, it will stop routine and retry if the stopCh is closed.